tests/api_test.py

```diff
@@ -2666,8 +2666,7 @@ def test_escaped_tracer_transform_name(self):
       _ = self._saved_tracer+1
 
   def test_escaped_tracer_shape_dtype(self):
-    with self.assertRaisesRegex(core.UnexpectedTracerError,
-                                r"shape \(4, 3\) and dtype int32"):
+    with self.assertRaisesRegex(core.UnexpectedTracerError, r"int32\[4,3\]"):
```

Comment:
I changed this test because I improved the error message. I changed the error message in this PR because previously the error message formatting always attempted to print the shape and dtype associated with a Tracer, but that Tracer could be on a core.unit (which has no shape or dtype).